src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -1381,20 +1381,25 @@ void M(bool b)
     }
 }");
 
-            // Implicit pointer conversions are not currently handled, ptr3/4 and ptr7/8 should warn: https://github.com/dotnet/roslyn/issues/39865
             comp.VerifyDiagnostics(
                 // (12,13): error CS0173: Type of conditional expression cannot be determined because there is no implicit conversion between 'delegate*<string>' and 'delegate*<string>'
                 //         _ = b ? ptr1 : ptr3;
                 Diagnostic(ErrorCode.ERR_InvalidQM, "b ? ptr1 : ptr3").WithArguments("delegate*<string>", "delegate*<string>").WithLocation(12, 13),
                 // (13,13): error CS0173: Type of conditional expression cannot be determined because there is no implicit conversion between 'delegate*<string>' and 'delegate*<string?>'
                 //         _ = b ? ptr1 : ptr4;
                 Diagnostic(ErrorCode.ERR_InvalidQM, "b ? ptr1 : ptr4").WithArguments("delegate*<string>", "delegate*<string?>").WithLocation(13, 13),
+                // (14,24): warning CS8619: Nullability of reference types in value of type 'delegate*<string?>' doesn't match target type 'delegate*<string>'.
```

Comment:
>delegate*<string> [](start = 140, length = 17)

Shouldn't we infer ```delegate*<string?>``` instead?

Reply:
No, these are `ref`s. Unfortunately, our CSharpErrorMessageFormat does not include return ref kinds, so the error message isn't super clear. I could change that to include them, but I'm unsure of the fully ramifications of such a change.

---
In reply to: [426093508](https://github.com/dotnet/roslyn/pull/44257#discussion_r426093508) [](ancestors = 426093508)

Reply:
> Unfortunately, our CSharpErrorMessageFormat does not include return ref kinds, so the error message isn't super clear. 

Since this isn't really a member, perhaps the symbol display for function pointers shouldn't be driven by the options for members and should always display ```ref``` kind because this is important aspect of the type. Same for other things we might be dropping  

---
In reply to: [426844096](https://github.com/dotnet/roslyn/pull/44257#discussion_r426844096) [](ancestors = 426844096,426093508)

Reply:
I've added a note to the tracking issue to consider that change, will resolve this thread.

---
In reply to: [426910541](https://github.com/dotnet/roslyn/pull/44257#discussion_r426910541) [](ancestors = 426910541,426844096,426093508)